### PR TITLE
Feature/reorganize ingredient data

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
the 4th task in Migrate old ingredients into new "objects dict" #57 

Solution
========
Created a partners object to convert ingredients that should be moved into this object. 
Wrote tests for _convert_to_objects   

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: cellpack/autopack/loaders/recipe_loader.py [hunk 1]
```diff
@@ -119,10 +107,49 @@ def _request_sub_recipe(self, inode):
...
+                representations["atomic"]["transform"] = old_ingredient["source"][
+                    "transform"
+                ]
+
+        return representations
+
+    @staticmethod
+    def _convert_to_partners(old_ingredient):
+        partners = {}
+        for attribute in old_ingredient:
```
Comment (meganrm) [inline cellpack/autopack/loaders/recipe_loader.py:149]: Even though this isn't a huge amount of data it's generally better to not loop through data more that once if you don't need to. In line 157 we are already looping over all the attributes

File: cellpack/autopack/loaders/recipe_loader.py [hunk 2]
```diff
@@ -132,6 +159,27 @@ def _migrate_ingredient(old_ingredient):
                 new_ingredient[v1_to_v2_name_map[attribute]] = old_ingredient[attribute]
             elif attribute in unused_attributes_list:
                 del old_ingredient[attribute]
+
```
Comment (meganrm) [inline cellpack/autopack/loaders/recipe_loader.py:162]: Heres where you could move line 150. 
```python
if attribute in convert_to_partners_map:
    if "partners" not in new_ingredient:
         partners = {}
         new_ingredient["partners"] = partners
    partners[convert_to_partners_map[attribute]] = old_ingredient[attribute]

File: cellpack/tests/test_recipe_version_migration.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+                "nbMol": 15,
+                "encapsulatingRadius": 100,
+                "orientBiasRotRangeMin": 6,
+                "orientBiasRotRangeMax": 12,
+            },
+            {
+                "count": 15,
+                "orient_bias_range": [6, 12],
+                "representations": RecipeLoader.default_values["representations"],
+                "partners": {},
```
Comment (meganrm) [inline cellpack/tests/test_recipe_version_migration.py:47]: If there is no partners data we don't want this field at all

File: cellpack/autopack/loaders/recipe_loader.py [hunk 4]
```diff
@@ -132,6 +151,31 @@ def _migrate_ingredient(old_ingredient):
...
+                old_ingredient["orientBiasRotRangeMin"]
+                if "orientBiasRotRangeMin" in old_ingredient
+                else -pi
+            )
+            range_max = (
+                old_ingredient["orientBiasRotRangeMax"]
+                if "orientBiasRotRangeMax" in old_ingredient
+                else pi
+            )
+            new_ingredient["orient_bias_range"] = [range_min, range_max]
```
Comment (meganrm) [inline cellpack/autopack/loaders/recipe_loader.py:174]: you could change this to a function that returns  `[range_min, range_max]`

ie:
```python
def _convert_rotation_range(old_ingredient):
            range_min = (
                old_ingredient["orientBiasRotRangeMin"]
                if "orientBiasRotRangeMin" in old_ingredient
                else -pi
            )
            range_max = (
                old_ingredient["orientBiasRotRangeMax"]
                if "orientBiasRotRangeMax" in old_ingredient
                else pi
            )
            return [range_min, range_max]

# lines 164 - 174:
new_ingredient["orient_bias_range"] = RecipeLoader._convert_rotation_range(old_ingredient)
```

**Review by meganrm — APPROVED** [review]
Looks great! really nice job on this

**Comment by rugeli** [comment]
merged to staging/ and deleted feature/ branch, grateful for helps!

**Pull request merged**